api/root.go

```diff
@@ -0,0 +1,66 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/jackc/pgx/v4"
+)
+
+type proofLookup struct {
+	Proof []string `json:"proof"`
+}
+
+func proofURLToDBQuery(param string) string {
+	lookup := proofLookup{
+		Proof: strings.Split(param, ","),
+	}
+
+	q, err := json.Marshal([]proofLookup{lookup})
+	if err != nil {
+		return ""
+	}
+
+	return string(q)
+}
+
+type rootResp struct {
+	Root hexutil.Bytes `json:"root"`
+}
+
+func (s *Server) GetRoot(w http.ResponseWriter, r *http.Request) {
+	var (
+		ctx     = r.Context()
+		proof   = r.URL.Query().Get("proof")
+		dbQuery = proofURLToDBQuery(proof)
+	)
+	if proof == "" || dbQuery == "" {
+		s.sendJSONError(r, w, nil, http.StatusBadRequest, "missing list of proofs")
+		return
+	}
+
+	q := `
+		SELECT root
+		FROM trees
+		WHERE proofs @> $1
```

Comment:
@ryandotsmith btw, when i ran `explain analyze` on this, it appears to be doing a seq scan:

```
Seq Scan on trees  (cost=0.00..34.74 rows=1 width=33) (actual time=10.943..61.470 rows=1 loops=1)
  Filter: (proofs @> '[{"proof": ["0x2cac63703751b7aa098953eb49a5106f602471eeb779aa4a82a744310d75be70", "0x9b1177ebb4a06fc44f01964b3d9ae7b3f21950259f8833ea52b6a2b7a1fcb71d", "0xcc7d1192a8efc9c8a15d78e5320086cb14ca62015725ba94e467d03af834c400"]}]'::jsonb)
  Rows Removed by Filter: 298
Planning Time: 1.361 ms
Execution Time: 61.536 ms
```

should we make an index here?